document/execution/numerics.rst

```diff
@@ -1,54 +1,1472 @@
+   \end{array}
+
+
+.. _op-fnearest:
+
+:math:`\fnearest_N(z)`
+......................
+
+* If :math:`z` is a NaN, then return an element of :math:`\nan_N\{z\}`.
+
+* Else if :math:`z` is an infinity, then return :math:`z`.
+
+* Else if :math:`z` is a zero, then return :math:`z`.
+
+* Else if :math:`z` is greater than :math:`0` but smaller than or equal to :math:`0.5`, then return positive zero.
+
+* Else if :math:`z` is smaller than :math:`0` but greater than or equal to :math:`-0.5`, then return negative zero.
+
+* Else return the integral value that is nearest to :math:`z`; if two values are equally near, return the even one.
+
+.. math::
+   \begin{array}{@{}lcll}
+   \fnearest_N(\pm \NAN(n)) &=& \nan_N\{\pm \NAN(n)\} \\
+   \fnearest_N(\pm \infty) &=& \pm \infty \\
+   \fnearest_N(\pm 0) &=& \pm 0 \\
+   \fnearest_N(+ q) &=& +0 & (0 < +q \leq 0.5) \\
+   \fnearest_N(- q) &=& -0 & (-0.5 \leq -q < 0) \\
+   \fnearest_N(\pm q) &=& \ieee_N(\pm i) & (|i - q| < 0.5) \\
+   \fnearest_N(\pm q) &=& \ieee_N(\pm i) & (|i - q| = 0.5 \wedge i~\mbox{even}) \\
+   \end{array}
+
+
+.. _op-feq:
+
+:math:`\feq_N(z_1, z_2)`
+........................
+
+* If either :math:`z_1` or :math:`z_2` is a NaN, then return :math:`0`.
+
+* Else if both :math:`z_1` and :math:`z_2` are zeroes, then return :math:`1`.
+
+* Else if both :math:`z_1` and :math:`z_2` are the same value, then return :math:`1`.
+
+* Else return :math:`0`.
+
+.. math::
+   \begin{array}{@{}lcll}
+   \feq_N(\pm \NAN(n), z_2) &=& 0 \\
+   \feq_N(z_1, \pm \NAN(n)) &=& 0 \\
+   \feq_N(\pm 0, \mp 0) &=& 1 \\
+   \feq_N(z, z) &=& 1 \\
+   \feq_N(z_1, z_2) &=& 0 \\
+   \end{array}
+
+
+.. _op-fne:
+
+:math:`\fne_N(z_1, z_2)`
+........................
+
+* If either :math:`z_1` or :math:`z_2` is a NaN, then return :math:`0`.
+
+* Else if both :math:`z_1` and :math:`z_2` are zeroes, then return :math:`0`.
+
+* Else if both :math:`z_1` and :math:`z_2` are the same value, then return :math:`0`.
+
+* Else return :math:`1`.
+
+.. math::
+   \begin{array}{@{}lcll}
+   \fne_N(\pm \NAN(n), z_2) &=& 0 \\
+   \fne_N(z_1, \pm \NAN(n)) &=& 0 \\
+   \fne_N(\pm 0, \mp 0) &=& 0 \\
+   \fne_N(z, z) &=& 0 \\
+   \fne_N(z_1, z_2) &=& 1 \\
+   \end{array}
+
+
+.. _op-flt:
+
+:math:`\flt_N(z_1, z_2)`
+........................
+
+* If either :math:`z_1` or :math:`z_2` is a NaN, then return :math:`0`.
+
+* Else if :math:`z_1` and :math:`z_2` are the same value, then return :math:`0`.
+
+* Else if :math:`z_1` is negative infinity, then return :math:`1`.
+
+* Else if :math:`z_2` is positive infinity, then return :math:`1`.
+
+* Else if both :math:`z_1` and :math:`z_2` are zeroes, then return :math:`0`.
+
+* Else if :math:`z_1` is smaller than :math:`z_2`, then return :math:`1`.
+
+* Else return :math:`0`.
+
+.. math::
+   \begin{array}{@{}lcll}
+   \flt_N(\pm \NAN(n), z_2) &=& 0 \\
+   \flt_N(z_1, \pm \NAN(n)) &=& 0 \\
+   \flt_N(z, z) &=& 0 \\
+   \flt_N(- \infty, z_2) &=& 1 \\
```

Comment:
why do you add special handling for \flt_N(- \infty, z_2) but not for \flt_N(z_1, - \infty)?